### PR TITLE
[KUNLUN] add op: maxpool_with_index

### PR DESCRIPTION
**Pull request by XiaociZhang**
### PR types

Others
### PR changes

Others
### Describe

KL support op 'maxpool with index'

### CODE REVIEW

File: paddle/phi/kernels/xpu/pool_grad_kernel.cc [hunk 1]
```diff
@@ -142,6 +142,79 @@ void Pool2dGradKernel(const Context& ctx,
...
+      input_grad,
+      n,
+      c,
+      in_h,
+      in_w,
+      ksize,
+      strides,
+      paddings,
+      true);
+  PADDLE_ENFORCE_EQ(
```
Comment (QingshuChen) [inline paddle/phi/kernels/xpu/pool_grad_kernel.cc]: 用PADDLE_ENFORCE_XDNN_SUCCESS这个宏

File: paddle/phi/backends/xpu/xpu2_op_list.cc [hunk 2]
```diff
@@ -342,6 +342,10 @@ XPUOpMap& get_kl2_ops() {
                      phi::DataType::BOOL,
                      phi::DataType::FLOAT16,
                      phi::DataType::FLOAT32})},
+      {"max_pool2d_with_index",
```
Comment (QingshuChen) [inline paddle/phi/backends/xpu/xpu2_op_list.cc:376]: 没有单测

File: paddle/phi/kernels/xpu/pool_grad_kernel.cc (hunk 1 above)
Comment (XiaociZhang) [reply paddle/phi/kernels/xpu/pool_grad_kernel.cc]: ok

File: paddle/phi/backends/xpu/xpu2_op_list.cc (hunk 2 above)
Comment (XiaociZhang) [reply paddle/phi/backends/xpu/xpu2_op_list.cc:376]: ok

**Review by QingshuChen — APPROVED** [review]

**Pull request merged**